ibis/expr/types/generic.py

```diff
@@ -851,9 +851,13 @@ def __hash__(self) -> int:
         return super().__hash__()
 
     def __eq__(self, other: Value) -> ir.BooleanValue:
+        if other is None:
+            return _binop(ops.IdenticalTo, self, other)
```

Comment:
We currently allow `None` in equals and not equals expressions, like `a == None` to mean `a is None`. This was being turned into `a is None` by sqlalchemy, here we are moving that transformation into the ibis API.

Reply:
I think this should be handled at the backend instead. 

Is it well defined what is the difference between `EqualTo` and `IdenticalTo`?

Reply:
Yes. `Equals` is not null safe, `IdenticalTo` is null safe.